backends-clickhouse/src/test/scala/org/apache/spark/sql/execution/benchmarks/CHAggAndShuffleBenchmark.scala

```diff
@@ -0,0 +1,444 @@
+      .setIfMissing("spark.memory.offHeap.size", offheapSize)
+      .setIfMissing("spark.sql.columnVector.offheap.enabled", "true")
+      .setIfMissing("spark.sql.shuffle.partitions", shufflePartition)
+      .setIfMissing("spark.driver.memory", memorySize)
+      .setIfMissing("spark.executor.memory", memorySize)
+      .setIfMissing("spark.sql.files.maxPartitionBytes", "1G")
+      .setIfMissing("spark.sql.files.openCostInBytes", "1073741824")
+      .setIfMissing("spark.gluten.sql.columnar.coalesce.batches", "true")
+      .setIfMissing("spark.shuffle.manager", "sort")
+      .setIfMissing("spark.io.compression.codec", "SNAPPY")
+
+    val sparkSession = SparkSession.builder.config(conf).getOrCreate()
+    sparkSession.sparkContext.setLogLevel("WARN")
+    sparkSession
+  }
+
+  override def runBenchmarkSuite(mainArgs: Array[String]): Unit = {
+    val (
+      fileFormat,
+      parquetDir,
+      executedCnt,
+      executedVanilla,
+      finalOutput,
+      groupByKeys,
+      aggFunctions,
+      filters) =
+      if (mainArgs.isEmpty) {
+        (
+          "parquet",
+          "/data/tpch-data/parquet/lineitem/",
+          3,
+          true,
+          "min(min_partkey), min(min_suppkey), min(sum_base_price)",
+          "l_partkey",
+          "min(l_partkey) as min_partkey,min(l_suppkey) as min_suppkey," +
+            "sum(l_extendedprice) AS sum_base_price",
+          "WHERE l_shipdate <= date'1996-09-02' - interval 1 year")
+      } else {
+        if (mainArgs.length == 4) {
+          (
+            mainArgs(0),
+            mainArgs(1),
+            mainArgs(2).toInt,
+            mainArgs(3).toBoolean,
+            "min(min_partkey), min(min_suppkey), min(sum_base_price)",
+            "l_partkey",
+            "min(l_partkey) as min_partkey,min(l_suppkey) as min_suppkey," +
+              "sum(l_extendedprice) AS sum_base_price",
+            "WHERE l_shipdate <= date'1996-09-02' - interval 1 year")
+        } else {
+          (
+            mainArgs(0),
+            mainArgs(1),
+            mainArgs(2).toInt,
+            mainArgs(3).toBoolean,
+            mainArgs(4),
+            mainArgs(5),
+            mainArgs(6),
+            mainArgs(7))
+        }
+      }
+
+    def allStages = spark.sql(s"""
+                                 | SELECT $finalOutput
+                                 | FROM (
+                                 | SELECT
+                                 |     $groupByKeys
+                                 |     ${if (aggFunctions.nonEmpty) "," + aggFunctions else ""}
+                                 | FROM
+                                 |     $fileFormat.`$parquetDir`
+                                 | $filters
+                                 | GROUP BY
+                                 |     $groupByKeys) as a
+                                 |""".stripMargin)
+
+    val executedPlan = allStages.queryExecution.executedPlan
```

Comment:
It looks good, but it's better to add more comments to describe the stages of this SQL

Reply:
Done.

Reply:
Thanks!